regression/contracts/loop_assigns-03/test.desc

```diff
@@ -0,0 +1,17 @@
+CORE
+main.c
+--apply-loop-contracts
+^EXIT=10$
+^SIGNAL=0$
+^\[main.\d+\] .* Check loop invariant before entry: SUCCESS$
+^\[main.\d+\] .* Check that i is assignable: FAILURE$
+^\[main.\d+\] .* Check that b->data\[(.*)i\] is assignable: SUCCESS$
+^VERIFICATION FAILED$
+--
+--
+This test (taken from #6021) shows the need for assigns clauses on loops.
+The alias analysis in this case returns `unknown`,
+and so we must manually annotate an assigns clause on the loop.
+
+Note that the annotated assigns clause in this case is an underapproximation,
```

Comment:
What is the consequence of this?  Does it produce unsound results or simply inefficient ones?

Reply:
Previously CBMC used to crash (see: #6021) because it couldn't resolve the `unknown` to an underlying object.

After merging this PR, the user would receive an error message suggesting them to add an assigns clause on the loop.

Reply:
If the error message is the "Check that i is assignable" VC failure then, OK, that is fine.  If there is something else then that should be checked for.

Reply:
Oops. I'm sorry, I was thinking of something else (not providing the assigns clause at all), which is unrelated to your question.

Yes, the expected error here is `Check that i is assignable: FAILURE`.

Reply:
Also, it wouldn't lead to unsound results, but raise a verification failure.

Reply:
Thanks for your thoughts; that is fine.